tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -896,6 +896,8 @@ where
             {
                 let mut writer = self.make_writer.make_writer_for(event.metadata());
                 let _ = io::Write::write_all(&mut writer, buf.as_bytes());
+            } else {
+                eprintln!("[tracing-subscriber] Unable to format the following event: {:?}", event);
```

Comment:
take it or leave it: looking at the example output you posted, i notice that the way the event is formatted using `fmt::Debug` is quite a lot of text. perhaps it would be better to print something like this:
```suggestion
                eprintln!("[tracing-subscriber] Unable to format {} with {:?}", event.metadata().name(), event.fields());
```
which would be a bit more concise but still include relevant information about the event.

Reply:
Yeah, I think that makes sense, as long as the user can still find where/which actual event is the one that is failing, that should be enough info.

Reply:
Addressed in my second commit